### PR TITLE
feat(survey): Support identifying iterations of a repeatable survey

### PR DESCRIPTION
**Pull request by Phanatic**
## Changes

As part of https://github.com/PostHog/posthog/issues/22099, we'd like to uniquely identify each "iteration" of a survey that a user is interacting with. 
1. Change the localStorage key used to store the `seenSurvey` key to have the iteration as the suffix. 
2. Send the `current_iteration` and `current_iteration_start_date` properties on the event
3. Moved `closeSurveyPopup` into the `survey-utils.tsx` file.
4. Rename `closeSurveyPopup` to `dismissedSurveyEvent` to match `sentSurveyEvent`

## Checklist
- [x] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [x] Accounted for the impact of any changes across different browsers
- [x] Accounted for backwards compatibility of any changes (no breaking changes in posthog-js!)

### CODE REVIEW

**Review by neilkakkar — COMMENTED** [review]
this seems reasonable 👍 , waiting for the main posthog update completion so I can test once e2e before approving

File: src/extensions/surveys/surveys-utils.tsx [hunk 1]
```diff
@@ -604,6 +625,24 @@ export const getDisplayOrderQuestions = (survey: Survey): SurveyQuestion[] => {
     return reverseIfUnshuffled(survey.questions, shuffle(survey.questions))
 }
 
+export const getSurveySeenKey = (survey: Survey): string => {
+    let surveySeenKey = `seenSurvey_${survey.id}`
+    if (survey.current_iteration && survey.current_iteration > 0) {
+        surveySeenKey = `seenSurvey_${survey.id}_${survey.current_iteration}`
+    }
```
Comment (dmarticus) [inline src/extensions/surveys/surveys-utils.tsx:632]: camelCase and snake_case?  Is this a pattern for tracking metadata after variables?  Reads a bit odd to me but I could just be overthinking it.

**Review by dmarticus — COMMENTED** [review]
one comment, but not blocking.

File: src/extensions/surveys/surveys-utils.tsx (hunk 1 above)
Comment (Phanatic) [reply src/extensions/surveys/surveys-utils.tsx:632]: yeah, its the pattern used for these sorts of keys. original feedback on this variable is in https://github.com/PostHog/posthog-js/pull/1200#discussion_r1618758364

**Review by dmarticus — APPROVED** [review]
:shipit:

**Review by dmarticus — APPROVED** [review]
:shipit:

**Pull request merged**